File: gradletrans/gson/ios/ios/ViewController.m
//
//  ViewController.m
//  ios
//
//  Created by Kevin Galligan on 6/11/16.
//  Copyright © 2016 Kevin Galligan. All rights reserved.
//

#import "ViewController.h"
#import "AllTests.h"
#import "java/lang/System.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)runClicked:(id)sender {
    
    
    [AllTests runTests];
    //    [_benchmarkPresenter startBenchmark];
    //    [_simpleTestPresenter runDbTest];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: mvp/DoppelRace/DoppelRace/DoppelRace/ViewController.swift
//
//  ViewController.swift
//  DoppelRace
//
//  Created by Kevin Galligan on 4/13/16.
//  Copyright © 2016 Kevin Galligan. All rights reserved.
//

import UIKit


class ViewController: UIViewController, DRHost {

    var racePresenter : DRRacePresenter?;
    @IBOutlet weak var messageOutput: UILabel?
    @IBOutlet weak var progressBar: UIProgressView?
    
    required  init?(coder aDecoder: NSCoder) {
        AndroidOsLooper.prepareMainLooper();
        super.init(coder: aDecoder);
        racePresenter = DRRacePresenter(DRHost: self);
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        racePresenter!.startRaceWithInt(10)
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func getContext() -> AndroidContentContext{
        return AndroidContentIOSContext()
    }

    func postProgressWithCoTouchlabDoppelraceTasksRunRaceTask_RunRaceProgress(progress: CoTouchlabDoppelraceTasksRunRaceTask_RunRaceProgress){
        messageOutput!.text = progress.getMessage();
        progressBar!.progress = Float(progress.getPercentDone());
    }
    
    func raceDone(){
        messageOutput!.text = "Done!";
        progressBar!.progress = 1;
    }

}


